chameleon-core/src/test/java/org/hibernate/omm/crud/SimpleCRUDTests.java

```diff
@@ -120,6 +120,34 @@ void testUpdate() {
         });
     }
 
+    @Test
+    void testUpdateQuery() {
+        insertBook();
+        var newAuthor = "Fyodor Dostoevsky";
+        var newTitle = "Crime and Punishment";
+        int newPublishYear = 1866;
+
+        // the following JSON command will be issued:
+        // { update: "books", updates: [ { q: { _id: { $eq: ? } }, u: { $set: { author: ?, publishYear: ?, title: ? } } } ] }
+        sessionFactory.inTransaction(session -> {
+            var update = session.createMutationQuery("update Book " +
+                    "set author = :author, title = :title, publishYear = :publishYear " +
+                    "where id = :id");
+            update.setParameter("id", id);
+            update.setParameter("author", newAuthor);
+            update.setParameter("title", newTitle);
+            update.setParameter("publishYear", newPublishYear);
+            update.executeUpdate();
+        });
+        sessionFactory.inSession(session -> {
+            var query = session.createQuery("from Book where id = :id", Book.class);
+            query.setParameter("id", id);
+            var book = query.getSingleResult();
+            assertThat(book).extracting("author", "title", "publishYear")
+                    .containsOnly(newAuthor, newTitle, newPublishYear);
```

Comment:
I really love AssertJ for it has unbeatable readability and writing unit testing even becomes enjoyable. I advocated its usage in Hiberante team and Steve adopted it (so other team members adopted it naturally). Prior to that Hamcrest is heavily relied upon.